### PR TITLE
FIX missing filter by current contact

### PR DESCRIPTION
**Pull request by hregis**

### CODE REVIEW

**Comment by hregis** [comment]
to fix this issue https://github.com/Dolibarr/dolibarr/issues/11549

File: htdocs/contact/consumption.php [hunk 1]
```diff
@@ -300,6 +300,7 @@
 	$sql.= " FROM "/*.MAIN_DB_PREFIX."societe as s, "*/.$tables_from;
 //	if ($type_element != 'fichinter') $sql.= ' LEFT JOIN '.MAIN_DB_PREFIX.'product as p ON d.fk_product = p.rowid ';
 	$sql.= $where;
+	$sql.= ' AND ec.fk_socpeople = '.$object->id;
```
Comment (eldy) [inline htdocs/contact/consumption.php]: The table called "ec" was added in request with a INNER JOIN or a LEFT JOIN. This means you must not add condition in the global where but you have to put it after the "... ON ..." that is set just after the INNER or LEFT JOIN.

Comment (hregis) [reply htdocs/contact/consumption.php]: @eldy 
it is a tab that is in the contact card, should not we filter with the contact ID globally? This tab does not allow the contact to see only the documents with which they are associated?

Comment (hregis) [reply htdocs/contact/consumption.php]: @eldy 
ok i send a fix
maybe it would be good to mutualized SQL queries because many of them are duplicates

**Comment by hregis** [comment]
@eldy ping

**Pull request merged**